src/main/ruby/core/numeric_mirror.rb

```diff
@@ -47,15 +47,17 @@ def step_float_size(value, limit, step, asc)
         end
       end
 
-      def step_size(limit, step)
-        values = step_fetch_args(limit, step)
+      def step_size(limit, step, by)
```

Comment:
They weirdly aren't the same thing since `by` is treated more permissibly:
```ruby
0.step(0, by: 0).first
#=> 0

0.step(0, step = 0).first
#!> ArgumentError: step can't be 0
```